ships/0005-local-source-upload.md

```diff
@@ -0,0 +1,262 @@
+---
+title: local-source-upload
+authors:
+  - @otaviof
+reviewers:
+  - @alicerum
+  - @adambkaplan
+approvers:
+creation-date: 2021-06-23
+last-updated: 2021-06-23
+---
+
+# Source Upload
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Summary
+
+Proposal to improve developer-experience by allowing to upload a local repository clone to the
+Kubernetes cluster, and use Shipwright Build Controller to produce a container-image out of it.
+
+As a developer, I can try Shipwright before opening a pull-request, and take advantage of the cluster
+resources to create the image. Effectively moving shipwright closer to the developer's inner-loop, in
+other words closer to the frequent activities while writing code ([1][InnerLoopA], [2][InnerLoopB]),
+in other words we are "moving left before _shipping-right_".
+
+## Motivation
+
+Shipwright Build-Controller allows teams to build container-images out of Git repositories, register
+and track their application builds in a Kubernetes native environment. This enhancement proposal
+extends this ability for developers to upload uncommitted changes and use the same machinery during
+the inner-loop as well.
+
+As a developer you can try out changes before opening a pull-request, make sure the final
+container-image will work out as expected, and effectively improve the developer experience.
+
+The approach proposed here is complementary to existing [EP][LocalSourceCodeEP] detailing local
+source upload to a container-registry and later on self-extracting the image before the actual
+container build process starts.
+
+### Goals
+
+- Improve developer experience, local clone can produce a container-image using Shipwright machinery
+- Move Shipwright CLI closer to developer's inner-loop
+- Complementary to [upload-to-container-registry (EP)][LocalSourceCodeEP], uploading data directly to
+the cluster instead
+
+## Proposal
```

Comment:
It would be helpful to include under the `Proposal` section, how the usage of this feature will look in SHP cli. Keep in mind the following:

- This provides another plugable option to achieve local source code. The different flavors of local source code should be almost transparent for the user. See for example the idea for the initial local source code option in [ep](https://github.com/shipwright-io/build/blob/main/docs/proposals/enable-local-source-code-support.md#implementation-detailsnotesconstraints-optional).
- We should avoid having different subcommands for it, so we might wanna signalize the flavors of local source code under the `shp build create`. 

The above should be stated in here.

Reply:
Indeed. I'll add on the document itself, my suggestion is based on the steps you see on the [demo-video](https://youtu.be/btgHUUMfSTo).

Reply:
Just added the command-line usage example in the proposal itself, taking in consideration we have two different upload mechanisms, so we need a flag to differentiate them. Please consider.